x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/Version.java

```diff
@@ -86,26 +87,34 @@ public static Version fromString(String version) {
         // This is similar to how Elasticsearch's Build class digs up its build information.
         // Since version info is not critical, the parsing is lenient
         URL url = Version.class.getProtectionDomain().getCodeSource().getLocation();
-        String urlStr = url.toString();
+        CURRENT = extractVersion(url);
+    }
 
+    static Version extractVersion(URL url) {
+        String urlStr = url.toString();
         byte maj = 0, min = 0, rev = 0;
         String ver = "Unknown";
         String hash = ver;
-
-        if (urlStr.endsWith(".jar")) {
-            try (JarInputStream jar = new JarInputStream(url.openStream())) {
-                Manifest manifest = jar.getManifest();
-                hash = manifest.getMainAttributes().getValue("Change");
-                ver = manifest.getMainAttributes().getValue("X-Compile-Elasticsearch-Version");
-                byte[] vers = from(ver);
-                maj = vers[0];
-                min = vers[1];
-                rev = vers[2];
+        
+        if (urlStr.endsWith(".jar") || urlStr.endsWith(".jar!/")) {
+            try {
+                URLConnection conn = url.openConnection();
+                conn.setUseCaches(false);
+                
+                try (JarInputStream jar = new JarInputStream(conn.getInputStream())) {
+                    Manifest manifest = jar.getManifest();
+                    hash = manifest.getMainAttributes().getValue("Change");
+                    ver = manifest.getMainAttributes().getValue("X-Compile-Elasticsearch-Version");
+                    byte[] vers = from(ver);
+                    maj = vers[0];
+                    min = vers[1];
+                    rev = vers[2];
+                }
             } catch (Exception ex) {
                 throw new IllegalArgumentException("Detected Elasticsearch JDBC jar but cannot retrieve its version", ex);
             }
```

Comment:
Should we add a `finally` block to close the `conn` ?
(Maybe JarInputStream 's `close()` takes care of that, just wondering if we should be extra safe here)

Reply:
I don't think it's needed to close the URLConnection (actually, the class doesn't have a method to close/terminate it), only its stream: https://docs.oracle.com/javase/tutorial/networking/urls/readingWriting.html